spec/models/user_spec.rb

```diff
@@ -2799,6 +2800,29 @@ def reset_last_seen_cache!(user)
     end
   end
 
+  describe "#grouped_unread_high_priority_notifications" do
+    it "returns a map of high priority types to their unread count" do
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: true, read: true)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: true, read: false)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: false, read: true)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: false, read: false)
+
+      Fabricate(:notification, user: user, notification_type: 2, high_priority: true, read: false, topic: nil)
+
+      Fabricate(:notification, user: user, notification_type: 3, high_priority: true, read: false).tap do |n|
+        n.topic.trash!(Fabricate(:admin))
+      end
+
+      Fabricate(:notification, user: user, notification_type: 3, high_priority: false, read: true)
+
+      Fabricate(:notification, notification_type: 4, high_priority: true, read: false)
```

Comment:
Just by reading the method and this notification fabrication, I would expect it to be included in the results since it is high priority and unread.

Reply:
It's not included because it's for another user (note the absence of `user: user`).

Reply:
For readability sake, I think it'll be better if we can use an unused variable for the notifications here or maybe even just leaving a comment. Otherwise, it is going to be hard for someone else to understand what we're trying to fabricate here in the future.

Reply:
done!